cnxepub/models.py

```diff
@@ -128,12 +128,12 @@ def model_to_tree(model, title=None, lucent_id=TRANSLUCENT_BINDER_ID):
         {'id': <id>|'subcol', 'title': <title>, 'contents': [<tree>, ...]}
 
     """
-    if type(model) is TranslucentBinder:
+    id = model.ident_hash
+    if id is None and isinstance(model, TranslucentBinder):
```

Comment:
re: the specific code, I know how it evaluates - binders are a subclass of translucent binders. The previous code expected TLs to not have an id, so create one here so that the tree does not have id=None nodes for that case. It would have left a top level node for a putative un-identifed book as that id=None. If we're going to support one, why not the other? Also, passing "None" around feels wrong to me - bit of a code smell - see all the cleanup in the around required metadata in the tests. IT feels like mis-labelling a non-id'ed book as a subcol may be the lesser sin.